backend/app/api/endpoints/knowledge.py

```
@@ -468,6 +487,61 @@ def create_knowledge_base(
+
+    The requester must be able to read the repository, so that a wiki cannot be built
+    for a private repository they have no access to. Reading the resulting wiki is
+    then governed by knowledge-base permissions alone: place it in an organization
+    namespace to make it readable by everyone signed in, or keep it in a restricted
+    namespace and share it explicitly.
+    """
+    try:
+        source = SourceRepository.from_url(data.source_type, data.source_url)
+        assert_user_can_read_source(db, current_user.id, source)
+    except SourceAccessDenied as e:
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail=str(e))
```

Ruff (0.16.0)

[warning] Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)